R/write.R

```diff
@@ -1122,10 +1142,21 @@ write_formula <- function(
     formula <- "cm_formula"
   }
 
-  class(dfx$X) <- c("character", formula)
-  if (!is.null(dims)) {
-    if (array || cm) {
+  class(dfx$X) <- c(formula, "character")
+
+  if (is.null(dims)) {
+    dims <- wb_dims(start_row, start_col)
+  }
+
+  if (array || cm) {
+    if (length(dfx$X) == 1) {
       attr(dfx, "f_ref") <- dims
+    } else {
+      rowcols <- dims_to_rowcol(dims)
+      ddims   <- wb_dims(x = dfx$X, from_col = rowcols[[1]], from_row = as.integer(rowcols[[2]]))
```

Comment:
@olivroy I was wondering, if we should add something like: `wb_dims(from_dim = "A1")`.

Or even `wb_dims(from_dims = "A1:D2")` and additional positioning arguments like `right`, `below`, `top`, `left`. I have quite a few real life cases where I have to position various data frames next to each other on a single sheet, where I could envision this as useful. Might even be a integer argument, so that `right = 2` would create a dims range two columns next to the original.

```R
dims <- wb_dims(x = mtcars)
dims
#> "A1:K33"

wb_dims(from_dims = dims, x = mtcars, right = 2)
#> "M1:W33"

wb_dims(from_dims = dims, x = mtcars, below = 2)
#> "A35:K67"
```

Reply:
That would be lovely!